src/main/java/org/folio/support/ExpirationTool.java

```diff
@@ -103,74 +154,103 @@ private static Request changeRequestStatus(Request request) {
     return request;
   }
 
-  private static Request eraseRequestPosition(Request request) {
-    return request.withPosition(null);
-  }
+  private static Future<Void> reorderRequests(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                              String tenant, List<Request> requests) {
+
+    if (requests.isEmpty()) {
+      return succeededFuture();
+    }
 
-  private static Future<CompositeFuture> reorderRequests(Vertx vertx, String tenant, List<Request> requests) {
+    Map<String, List<Request>> groupedRequests = requests.stream()
+      .collect(Collectors.groupingBy(Request::getItemId));
 
-    return resetPositionsForOpenRequests(vertx, tenant)
-      .compose(v -> reorderRequestsForEachItem(vertx, tenant, requests));
+    return resetPositionsForOpenRequestsByIdemIds(conn, vertx, tenant, groupedRequests.keySet())
+      .compose(v -> reorderRequestsForEachItem(conn, vertx, tenant, groupedRequests));
   }
 
-  private static Future<CompositeFuture> reorderRequestsForEachItem(Vertx vertx, String tenant, List<Request> requests) {
+  private static Future<Void> reorderRequestsForEachItem(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                                         String tenant, Map<String, List<Request>> groupedRequests) {
 
-    Map<String, List<Request>> map = requests.stream()
-      .collect(Collectors.groupingBy(Request::getItemId));
+    Future<Void> future = succeededFuture();
+    for (Map.Entry<String, List<Request>> entry : groupedRequests.entrySet()) {
+      future = future.compose(v -> updateRequestsPositions(conn, vertx, tenant, entry.getValue()));
+    }
 
-    return CompositeFuture.all(map.entrySet()
-      .stream()
-      .map(entry -> updateRequestsPositions(vertx, tenant, entry.getValue()))
-      .collect(Collectors.toList()));
+    return future;
   }
 
-  private static Future<CompositeFuture> updateRequestsPositions(Vertx vertx, String tenant, List<Request> requests) {
+  private static Future<Void> updateRequestsPositions(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                                      String tenant, List<Request> requests) {
 
     AtomicInteger pos = new AtomicInteger(1);
+    Future<Void> future = succeededFuture();
 
-    return CompositeFuture.all(requests.stream()
-      .map(request -> updateRequest(vertx, tenant, request.withPosition(pos.getAndIncrement())))
-      .collect(Collectors.toList()));
+    for (Request request : requests) {
```

Comment:
How does this know what the correct position is? How are the requests order during this step?

(Apologies, I probably should've spotted this during the original work, it isn't really part of this change)

Reply:
Or does this work because the affected requests are ordered by position and translating them to a map by item ID does not affect that ordering?

Reply:
At first requests which need reordering is grouped by itemId and than request reordering process is performed separately for each item.
Probably  I made a mistake here. Seems like requests list should be sorted by position before reordering.

Reply:
I've just made a change. Now request list is sorted by position before reordering and there is a guarantee that requests are in right order before setting new positions.

Reply:
Thank you